test/unit/storage/test_xml_requests.rb

```diff
@@ -12,6 +12,16 @@ def teardown
     Fog.unmock!
   end
 
+  def test_get_url_path_has_query_params
```

Comment:
Thank you for the tests ❤️

Reply:
I had an equivalent test for the JSON response, but I saw this error:

```
NoMethodError:         NoMethodError: undefined method `issuer' for nil:NilClass
            /Users/stanhu/github/fog-google/lib/fog/storage/google_json/utils.rb:30:in `host_path_query'
            /Users/stanhu/github/fog-google/lib/fog/storage/google_json/utils.rb:10:in `https_url'
            /Users/stanhu/github/fog-google/lib/fog/storage/google_json/requests/get_object_https_url.rb:9:in `get_object_https_url'
            /Users/stanhu/github/fog-google/lib/fog/storage/google_json/requests/get_object_url.rb:16:in `get_object_url'
            /Users/stanhu/github/fog-google/test/unit/storage/test_json_requests.rb:18:in `test_get_url_path_has_query_params'
```

I assume `client.issuer` comes from `Google::Auth::ServiceAccountCredentials#make_creds`, but I punted on further investigation.

Reply:
Actually, `@client` is `nil` in the test. Not sure where that's supposed to come from.

Reply:
Ok, I did some simple mocking. Not sure if it's ideal. Take a look.